nova/compute/resource_tracker.py

```diff
@@ -1292,15 +1293,22 @@ def _update_usage_from_instances(self, context, instances, nodename):
                 "auto-correction of allocations.")
 
         instance_by_uuid = {}
+        for instance in instances:
+            instance_by_uuid[instance.uuid] = instance
+
+        bdms_by_uuid = (
+            objects.BlockDeviceMappingList.get_by_instance_uuids(
+                context, instance_by_uuid.keys()))
```

Comment:
As I read the code, this returns a `BlockDeviceMappingList` and not a dictionary. Can you point me to the place, where the grouping by instance-uuid is happening?

Reply:
Sorry, for some reason that was old code, the function called is `BlockDeviceMappingList.bdms_by_instance_uuid`.